### DISCUSSION TITLE
Welcome to MounteaDialogueSystem Discussions!

### DISCUSSION THREAD

**Discussion by pavlicekdominik** [Mountea-Framework/MounteaDialogueSystem#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by DrewdoesGames** [comment]
I've been working on my open world game for about 3 months, with over a year in UE5 (still a newbie), and have a pretty good timeline listed out of when I want things to be working for a pre-alpha state. It's a HUGE game, and a HUGE undertaking for a single person but it gives me a good hobby that challenges me. I enjoy problem solving, I hate doing the artsy stuff like animations and character designs so if anyone wants to work with me :)
I estimate I've got about a year of work left on this project before it's in a pre-alpha state and then I'd like to test it with friends, remove bugs, and hopefully *fingers crossed* launch a kick starter for it to higher an animator, a better character designer than myself, and ,depending on where the AI voice applications are at in a year, higher a voice actor or buy an AI voice App to do the dialogue. Until then, I've got a huge open world to design, layout, test, re-design, re-test, and the cycle continues. 

------------------------------------------------
Here I will make a list of things I've discovered how to do (that I could not find written in the documentation). For both my personal reference and anyone else that comes looking.

1. Inside of a Decorator you can in fact get the owner of the graph the decorator is attached to using "Get Owner" to first retrieve the 'Dialogue Participant Component' and from that do "Get Owning Actor".

I had tried this before and was attempting to first retrieve the graph and THEN get the owner, it drove me mad for about 30 minutes before I moved on so hopefully this saves someone else some headache. It might also be decorator dependent as I do not understand the difference between the decorators with a 'Star' and the ones with a 'Circle' as their Icon. 
Use case: validating decorators in real time AT the time the node is called to get the most up-to-date data from your actor.

-----------------------------------------------------------------------------------------------------------
2. When completing a dialogue if you want to use the dialogue system to prompt other things like shops, inventories, etc. that require a separate UI Widget you can use a "Send Command" decorator and send the command "UI" or something you prefer. Go to the actor that the node is calling as the participant, click on the 'Dialogue Participant Component', go to the right side in the details and scroll down to the events. Find the event "Participant Command Requested". Take the 'string' value and do a "Switch on string" node. Change one of the index to "UI" (whatever you typed in the command EXACTLY), and then: retrieve your player character --> get player state --> cast to (your player dialogue state name). Next, open your player dialogue state and create a variable "Keep UI". Next, back in your actor BP, from your casted player state set the newly created variable as 'TRUE'. Finally, go back to your 'Player Dialogue State' select the 'Dialogue Manager' and in the details on the right scroll down to the events and get "On Dialogue Closed" event, from the event: Branch if "KeepUI" --> from the FALSE branch get your player controller and "Set Input Mode Game Only". To reset the variable you can add the event "On Dialogue Initialized" and set the "KeepUI" variable to false.

Kind of long, might be an easier way to do it, but that's been my method throughout my project. Sometimes I find errors that I forgot to set the input but this has been my workaround and it's working well so far. If anything, hopefully it gives you an idea of how you can process it.
